### DISCUSSION TITLE
Getting extensions from the api response

### DISCUSSION THREAD

**Discussion by Kemosabert** [TanStack/query#3355] [Q&A] [answered]
Hey all,

We are building our GraphQL API and we started returning some metadata about a request in the response's `extensions` key as described by the GraphQL spec. It seems however that `react-query` only returns the data from the `data` and `error` keys from the response in the hooks. Is there a way to get the data returned by the `extensions` key in the hooks from `react-query` that we are overlooking in the docs?

If not, have you already considered adding this to the hooks? I am more than happy to help out with PR's for this if you are open to potentially adding this to the tool.

Thanks!

**Comment by TkDodo** [comment, marked as answer]
As always, react-query doesn't do anything in this regard. It takes the promise from the queryFn, awaits it and puts the data in the cache. What your promise contains is up to the library that you use to produce the promise. This might be `graphql-request`, or maybe `urql-core` or whatever you choose. If your queryFn returns `Promise.resolve(5)`, the number `5` will be in the cache :)

**Reply by Kemosabert** [reply]
Ah true, thanks a lot for your help.
